pgx-paths/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+/*
+Portions Copyright 2019-2021 ZomboDB, LLC.
+Portions Copyright 2021-2022 Technology Concepts & Design, Inc. <support@tcdi.com>
+
+All rights reserved.
+
+Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+*/
+
+use eyre::{eyre, WrapErr};
+use serde_json::value::Value as JsonValue;
+use std::{path::PathBuf, process::Command};
+
+pub fn get_target_dir() -> eyre::Result<PathBuf> {
+    let mut command = Command::new("cargo");
+    command
+        .arg("metadata")
+        .arg("--format-version=1")
+        .arg("--no-deps");
+    let output = command
+        .output()
+        .wrap_err("Unable to get target directory from `cargo metadata`")?;
+    if !output.status.success() {
+        return Err(eyre!(
+            "'cargo metadata' failed with exit code: {}",
+            output.status
+        ));
+    }
+
+    let json: JsonValue =
+        serde_json::from_slice(&output.stdout).wrap_err("Invalid `cargo metada` response")?;
+    let target_dir = json.get("target_directory");
+    match target_dir {
+        Some(JsonValue::String(target_dir)) => Ok(target_dir.into()),
+        v => Err(eyre!(
+            "could not read target dir from `cargo metadata` got: {:?}",
+            v,
+        )),
+    }
+}
```

Comment:
This function has a single consumer, in `cargo pgx`.

Reply:
I am also seeing it in `pgx-tests`. Maybe I am overlooking something:

![image](https://user-images.githubusercontent.com/387716/186253802-8593e53a-ec38-4e29-8c8a-82222b93dc71.png)

Reply:
Oh.
...Is my find not working?

Reply:
This is what I am finding with some `rg`:

```
$ rg "get_target_dir"
cargo-pgx/src/command/install.rs
17:use pgx_paths::{get_target_dir, versioned_so_name};
463:        .strip_prefix(get_target_dir()?.parent().unwrap())

cargo-pgx/src/command/package.rs
16:use pgx_paths::get_target_dir;
123:    let mut target_dir = get_target_dir()?;

cargo-pgx/src/command/schema.rs
21:use pgx_paths::get_target_dir;
180:    let mut target_dir_with_profile = get_target_dir()?;

cargo-pgx/src/command/test.rs
12:use pgx_paths::get_target_dir;
125:    let target_dir = get_target_dir()?;

pgx-paths/src/lib.rs
14:pub fn get_target_dir() -> eyre::Result<PathBuf> {

pgx-tests/src/framework.rs
16:use pgx_paths::get_target_dir;
261:        .env("CARGO_TARGET_DIR", get_target_dir()?);
528:    let mut target_dir = get_target_dir()?;

Reply:
Yeah. I think maybe the same thought about "this could also live in pgx-pg-config" still applies.